ci/builders/linux_framework_smoke.json

```diff
@@ -0,0 +1,70 @@
+{
+    "builds": [
+        {
+            "archives": [
+                {
+                    "name": "host_debug_unopt",
+                    "base_path": "out/host_debug_unopt/zip_archives/",
+                    "type": "gcs",
+                    "include_paths": [
+                        "out/host_debug_unopt/zip_archives/dart-sdk-linux-x64.zip"
+                    ],
+                    "realm": "production"
+                }
+            ],
+            "drone_dimensions": [
+                "device_type=none",
+                "os=Linux"
+            ],
+            "gn": [
+                "--unoptimized",
+                "--prebuilt-dart-sdk"
+            ],
+            "name": "host_debug_unopt",
+            "ninja": {
+                "config": "host_debug_unopt",
+                "targets": [
+                    "flutter/build/archives:dart_sdk_archive"
```

Comment:
I'm a bit confused about how this is working. The v1 recipe does a complete `host_debug_unopt` build, but this appears to be only building the Dart SDK archive.

The tests are using `FLUTTER_STORAGE_BASE_URL` to refer to the artifacts, but I would have thought that there wouldn't be a full set of artifacts there since they're not all getting built by this ninja step.

Maybe the tests in the framework repo only need the Dart SDK? If that's the case, then if there's a change in the framework repo which causes the tests below to need more engine artifacts, then that framework change will break the engine tree. To avoid that, I think the engine v2 recipe should match the engine v1 behavior, and build all of the Engine artifacts, and not just the Dart SDK archive.

Reply:
nvm, I see that widgets shard is already included.

Reply:
The dart-sdk-linux-x64.zip is downloaded when running [`flutter update-packages -v`](https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8779089292276869393/+/u/download_dependencies/stdout), before running the tests.

To confirm for v1 test behavior: do we know what artifacts these two test steps consume in addition to the `dart sdk`?
1) `flutter analyze --flutter-repo` (under path `flutter`)
2) `flutter test` (under path `flutter/packages/flutter`)

If there is no clear answer, does that mean we have to use the `--local-engine` for the v2 as we do not know what artifacts to upload to gcs? But `--local-engine` is not recommended for the v2 version per @godofredoc

Reply:
As one example, I'm surprised that these tests don't use `flutter_tester`. Maybe @goderbauer knows what artifacts these tests need?

Reply:
The related recipes includes how tests run:
1) flutter analyze will be running via `adhoc validation`, which runs [`dart --enable-asserts ./dev/bots/analyze.dart`](https://cs.opensource.google/flutter/recipes/+/main:recipe_modules/adhoc_validation/resources/analyze.sh)
2) `flutter test` will be running via `dev/bots/test.dart` in v2: https://flutter-review.googlesource.com/c/recipes/+/45400/5/recipes/engine_v2/tester.py#114

Reply:
> I would expect this to require the dart sdk and the sky_engine package for dart:ui. I am not sure, how the analyzer consumes dart:ui, though.

both `dart sdk` and `sky_engine` are downloaded from the newly built/uploaded artifacts (https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8779089292276869393/+/u/download_dependencies/stdout)

> This runs widget tests, so I would expect this to also need the flutter_tester binary.

The [log](https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8779089294121042961/+/u/Run_framework_tests_widgets_tests/stdout) shows all tests passed (ignore the warning), though we are not using `flutter_tester`. Does this mean it has been handled somewhere else, or it is not necessary?

> This runs widget tests

Actually this seems not the case. See [one log](https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8779015005170278385/+/u/Framework_test/test:_Framework_test/stdout) for v1 framework_smoke_tests, where unit tests under `cupertino`, `painting`, etc. are also executed. And these are covered under `libraries`.

Reply:
> both `dart sdk` and `sky_engine` are downloaded from the newly built/uploaded artifacts (https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8779089292276869393/+/u/download_dependencies/stdout)

If the build target and archive config only says to build and upload the Dart SDK, why does `sky_engine` also get built and uploaded?

Reply:
> The [log](https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8779089294121042961/+/u/Run_framework_tests_widgets_tests/stdout) shows all tests passed (ignore the warning), though we are not using `flutter_tester`. Does this mean it has been handled somewhere else, or it is not necessary?

I ran one of the tests in that log locally with `flutter test -v test/widgets/animated_padding_test.dart` and as the log indicates it is running the `flutter_tester` binary, see the `Starting flutter_tester process` line in the log below. So, from somewhere it is getting a flutter_tester binary.

<details><summary>Log</summary>

```
% flutter test -v test/widgets/animated_padding_test.dart                                                                                                       main
[   +7 ms] executing: [/Users/goderbauer/dev/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +61 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 42cf3e371a27b4882200b1f7ec7657e43a7cf249
[        ] executing: [/Users/goderbauer/dev/flutter/] git tag --points-at 42cf3e371a27b4882200b1f7ec7657e43a7cf249
[  +82 ms] Exit code 0 from: git tag --points-at 42cf3e371a27b4882200b1f7ec7657e43a7cf249
[   +2 ms] executing: [/Users/goderbauer/dev/flutter/] git describe --match *.*.* --long --tags 42cf3e371a27b4882200b1f7ec7657e43a7cf249
[  +85 ms] Exit code 0 from: git describe --match *.*.* --long --tags 42cf3e371a27b4882200b1f7ec7657e43a7cf249
[        ] 3.11.0-19.0.pre-21-g42cf3e371a
[   +3 ms] executing: [/Users/goderbauer/dev/flutter/] git symbolic-ref --short HEAD
[  +22 ms] Exit code 0 from: git symbolic-ref --short HEAD
[        ] main
[   +2 ms] executing: sw_vers -productName
[  +57 ms] Exit code 0 from: sw_vers -productName
[        ] macOS
[        ] executing: sw_vers -productVersion
[  +48 ms] Exit code 0 from: sw_vers -productVersion
[        ] 13.4
[        ] executing: sw_vers -buildVersion
[  +50 ms] Exit code 0 from: sw_vers -buildVersion
[        ] 22F66
[        ] executing: uname -m
[  +33 ms] Exit code 0 from: uname -m
[        ] x86_64
[  +13 ms] executing: sysctl hw.optional.arm64
[  +34 ms] Exit code 1 from: sysctl hw.optional.arm64
[        ] sysctl: unknown oid 'hw.optional.arm64'
[  +35 ms] executing: [/Users/goderbauer/dev/flutter/] git rev-parse --abbrev-ref --symbolic @{upstream}
[  +21 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{upstream}
[        ] origin/main
[        ] executing: [/Users/goderbauer/dev/flutter/] git ls-remote --get-url origin
[  +51 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] git@github.com:goderbauer/flutter.git
[  +24 ms] Found 1 files which will be executed as Widget Tests.
[  +11 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required, skipping update.
[   +5 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +92 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required, skipping update.
[        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[  +85 ms] Skipping pub get: version match.
[ +499 ms] running test package with arguments: [--chain-stack-traces, --, file:///Users/goderbauer/dev/flutter/packages/flutter/test/widgets/animated_padding_test.dart?]
00:00 +0: loading /Users/goderbauer/dev/flutter/packages/flutter/test/widgets/animated_padding_test.dart                                                                                             [ +349 ms] test 0: starting test /Users/goderbauer/dev/flutter/packages/flutter/test/widgets/animated_padding_test.dart
[   +9 ms] Discovered flutter_test_config.dart in /Users/goderbauer/dev/flutter/packages/flutter/test
[   +4 ms] Compiler will use the following file as its incremental dill file: /var/folders/s2/rdqjsnjx4vq97147345w30x80044f6/T/flutter_tools.Vcjb8k/flutter_test_compiler.Xqa4mu/output.dill
[        ] Listening to compiler controller...
[  +19 ms] Compiling file:///var/folders/s2/rdqjsnjx4vq97147345w30x80044f6/T/flutter_tools.Vcjb8k/flutter_test_listener.hRyEsZ/listener.dart
[  +90 ms] /Users/goderbauer/dev/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev /Users/goderbauer/dev/flutter/bin/cache/dart-sdk/bin/snapshots/frontend_server.dart.snapshot --sdk-root
/Users/goderbauer/dev/flutter/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --incremental --no-print-incremental-dependencies --target=flutter --experimental-emit-debug-metadata
-DFLUTTER_WEB_AUTO_DETECT=true --output-dill /var/folders/s2/rdqjsnjx4vq97147345w30x80044f6/T/flutter_tools.Vcjb8k/flutter_test_compiler.Xqa4mu/output.dill --packages
/Users/goderbauer/dev/flutter/packages/flutter/.dart_tool/package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --initialize-from-dill
/Users/goderbauer/dev/flutter/packages/flutter/build/test_cache/build/c075001b96339384a97db4862b8ab8db.cache.dill.track.dill --verbosity=error
[  +49 ms] <- compile file:///var/folders/s2/rdqjsnjx4vq97147345w30x80044f6/T/flutter_tools.Vcjb8k/flutter_test_listener.hRyEsZ/listener.dart
00:18 +0: loading /Users/goderbauer/dev/flutter/packages/flutter/test/widgets/animated_padding_test.dart                                                                                             [+18008 ms] <- accept
[        ] <- reset
[        ] Compiling file:///var/folders/s2/rdqjsnjx4vq97147345w30x80044f6/T/flutter_tools.Vcjb8k/flutter_test_listener.hRyEsZ/listener.dart took 18148ms
[   +1 ms] test 0: starting test device
[   +6 ms] test 0: awaiting connection to test device
[   +2 ms] test 0: VM Service uri is not available
[   +7 ms] test 0: test harness socket server is running at port:59126
[   +3 ms] Using this directory for fonts configuration: /var/folders/s2/rdqjsnjx4vq97147345w30x80044f6/T/flutter_tools.Vcjb8k/flutter_test_fonts.UjuSbt
[   +1 ms] test 0: Starting flutter_tester process with command=[/Users/goderbauer/dev/flutter/bin/cache/artifacts/engine/darwin-x64/flutter_tester, --disable-vm-service, --enable-checked-mode,
--verify-entry-points, --enable-software-rendering, --skia-deterministic-rendering, --enable-dart-profiling, --non-interactive, --use-test-fonts, --disable-asset-fonts,
--packages=/Users/goderbauer/dev/flutter/packages/flutter/.dart_tool/package_config.json, --flutter-assets-dir=/Users/goderbauer/dev/flutter/packages/flutter/build/unit_test_assets,
/var/folders/s2/rdqjsnjx4vq97147345w30x80044f6/T/flutter_tools.Vcjb8k/flutter_test_listener.hRyEsZ/listener.dart.dill], environment={FLUTTER_TEST: true, FONTCONFIG_FILE:
/var/folders/s2/rdqjsnjx4vq97147345w30x80044f6/T/flutter_tools.Vcjb8k/flutter_test_fonts.UjuSbt/fonts.conf, SERVER_PORT: 59126, APP_NAME: flutter, UNIT_TEST_ASSETS:
/Users/goderbauer/dev/flutter/packages/flutter/build/unit_test_assets}
[  +30 ms] test 0: Started flutter_tester process at pid 68559
00:19 +0: loading /Users/goderbauer/dev/flutter/packages/flutter/test/widgets/animated_padding_test.dart                                                                                             [+1704 ms] test 0: connected to test device, now awaiting test result
[        ] test 0: Waiting for test harness or tests to finish
00:21 +3: AnimatedPadding animated padding clamped to positive values                                                                                                                                [+1505 ms] test 0: Test harness is no longer needed by test process
[        ] test 0: finished
[   +1 ms] test 0: cleaning up...
[        ] test 0: ensuring test device is terminated.
[        ] test 0: Terminating flutter_tester process
[   +1 ms] test 0: Shutting down DevTools server
[   +1 ms] test 0: Test process is no longer needed by test harness
[   +3 ms] test 0: Shutting down test harness socket server
[  +15 ms] test 0: flutter_tester process at pid 68559 exited with code=-9
[        ] test 0: deleting temporary directory
[   +4 ms] test 0: finished
00:21 +3: All tests passed!                                                                                                                                                                          
[  +13 ms] Deleting /var/folders/s2/rdqjsnjx4vq97147345w30x80044f6/T/flutter_tools.Vcjb8k/flutter_test_compiler.Xqa4mu...
[   +4 ms] killing pid 68524
[ +100 ms] Deleting /var/folders/s2/rdqjsnjx4vq97147345w30x80044f6/T/flutter_tools.Vcjb8k/flutter_test_fonts.UjuSbt...
[   +4 ms] test package returned with exit code 0
[        ] Runtime for phase TestRunner: Wall-clock: 0:00:21.957568; combined: 0:00:21.957662.
[        ] Runtime for phase Compile: Wall-clock: 0:00:18.148791; combined: 0:00:18.148880.
[        ] Runtime for phase Run: Wall-clock: 0:00:03.260665; combined: 0:00:03.260669.
[        ] Runtime for phase CoverageTotal: Wall-clock: 0:00:00.000000; combined: 0:00:00.000000.
[        ] Runtime for phase CoverageCollect: Wall-clock: 0:00:00.000000; combined: 0:00:00.000000.
[        ] Runtime for phase CoverageParseJson: Wall-clock: 0:00:00.000000; combined: 0:00:00.000000.
[        ] Runtime for phase CoverageAddHitmap: Wall-clock: 0:00:00.000000; combined: 0:00:00.000000.
[        ] Runtime for phase CoverageDataCollect: Wall-clock: 0:00:00.000000; combined: 0:00:00.000000.
[        ] Runtime for phase WatcherFinishedTest: Wall-clock: 0:00:00.000851; combined: 0:00:00.000853.
[  +13 ms] "flutter test" took 22,668ms.
[ +277 ms] ensureAnalyticsSent: 252ms
[        ] Running 1 shutdown hook
[   +1 ms] Shutdown hooks complet
```

</details>

Reply:
Thanks @goderbauer for the explanation. Confirmed using led: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8778925221459687985/+/u/Run_framework_tests_widgets_tests/stdout

Same as `sky_engine`, the `flutter_tester` is downloaded via `linux-x64/artifacts.zip`, which is also built/uploaded in other targets.